### PR TITLE
Fixes unit tests runnings and improves eslint configuration

### PR DESCRIPTION
**Pull request by nzaytsev**
# Description

allows store __tests__ subdirectories
lints warnings
ignores trie tests
makes vscode-tests running with vscode/test-cli

# Checklist

- [x] I have followed the guidelines in the Contributing document
- [x] My changes follow the coding style of this project
- [x] My changes build without any errors or warnings
- [x] My changes have been formatted and linted
- [x] My changes include any required corresponding changes to the documentation (including CHANGELOG.md and README.md)
- [x] My changes have been rebased and squashed to the minimal number (typically 1) of relevant commits
- [x] My changes have a descriptive commit message with a short title, including a `Fixes $XXX -` or `Closes #XXX -` prefix to auto-close the issue that your PR addresses

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,12 +6,17 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/) and this p
 
 ## [Unreleased]
 
+### Fixed
+
+- Fixes vscode-test running Thanks to [PR #3391](https://github.com/gitkraken/vscode-gitlens/pull/3391) by Nikolay ([@nzaytsev](https://github.com/nzaytsev))
```
Comment (d13) [inline CHANGELOG.md:11]: We use this format for outside contributors. GL team members can just simplify that.

```suggestion
- Fixes running unit tests and improves eslint configuration
```

**Comment by eamodio** [comment]
Can we split this into an eslint change/upgrade (and take it further to ESLint 9 and the latest typescript-eslint) and then have the unit test work sit on top of that?

**Comment by eamodio** [comment]
Let's get this on top of https://github.com/gitkraken/vscode-gitlens/pull/3585

**Comment by nzaytsev** [comment]
closed in favor of https://github.com/gitkraken/vscode-gitlens/pull/3590

**Pull request closed without merging**